soh/soh/OTRGlobals.cpp

```diff
@@ -589,64 +612,90 @@ extern "C" SequenceData ResourceMgr_LoadSeqByName(const char* path)
 
 std::map<std::string, SoundFontSample*> cachedCustomSFs;
 
-extern "C" SoundFontSample* ResourceMgr_LoadAudioSample(const char* path) 
-{
-    if (std::string(path) == "")
+extern "C" SoundFontSample* ReadCustomSample(const char* path) {
+
+    if (!ExtensionCache.contains(path))
         return nullptr;
 
-    if (cachedCustomSFs.find(path) != cachedCustomSFs.end())
-        return cachedCustomSFs[path];
+    ExtensionEntry entry = ExtensionCache[path];
 
-    // Check if our file is actually a wav...
-    auto sampleRaw = OTRGlobals::Instance->context->GetResourceManager()->LoadFile(path);
+    auto sampleRaw = OTRGlobals::Instance->context->GetResourceManager()->LoadFile(entry.path);
     uint32_t* strem = (uint32_t*)sampleRaw->buffer.get();
     uint8_t* strem2 = (uint8_t*)strem;
 
-    if (strem2[0] == 'R' && strem2[1] == 'I' && strem2[2] == 'F' && strem2[3] == 'F') 
-    {
-        SoundFontSample* sampleC = (SoundFontSample*)malloc(sizeof(SoundFontSample));
+    SoundFontSample* sampleC = new SoundFontSample;
 
-        *strem++; // RIFF
-        *strem++; // size
-        *strem++; // WAVE
+    if (entry.ext == "wav") {
```

Comment:
We should probably not rely on the file extension and use the actual header in the file